src/interpreter/type_cast.rs

```diff
@@ -0,0 +1,224 @@
+            (PactType::any_try_into(&1_u16),  Ok(PactType::Numeric(Numeric(1)))),
+            (PactType::any_try_into(&2_u32),  Ok(PactType::Numeric(Numeric(2)))),
+            (PactType::any_try_into(&3_u64),  Ok(PactType::Numeric(Numeric(3)))),
+            (PactType::any_try_into(&4_u128), Ok(PactType::Numeric(Numeric(4)))),
+        ];
+        for (lhs, rhs) in tests {
+            assert_eq!(lhs, rhs);
+        }
+
+        // Assertion for overflow
+        assert_eq!(
+            PactType::any_try_into(&(core::u64::MAX as u128 + 2)),
+            Err(PactConversionErr::Overflow),
+        );
+    }
+
+    #[test]
+    fn it_converts_string_like() {
+        assert_eq!(
+            PactType::any_try_into(&"test"),
+            Ok(PactType::StringLike(StringLike(b"test"))),
+        );
+        assert_eq!(
+            PactType::any_try_into(&'a'.to_string()),
+            Ok(PactType::StringLike(StringLike(b"a"))),
+        );
+        assert_eq!(
+            PactType::any_try_into(&"test".to_string()),
+            Ok(PactType::StringLike(StringLike(b"test"))),
+        );
+
+        let v: Vec<u8> = vec![116, 101, 115, 116];
+        assert_eq!(
+            PactType::any_try_into(&v),
+            Ok(PactType::StringLike(StringLike(b"test")))
+        );
+
+        // Assertion for fixed hash types
+        let h32 = b"0x01";
+        let h64 = b"0x012345";
+        let h128 = b"0x01234567891011";
+        let h160 = b"0x012345678910111213";
+        let h256 = b"0x012345678910111213141516171819";
+        let h264 = b"0x0123456789101112131415161718192";
+        let h512 = b"0x01234567891011121314151617181920212223242526272829303132333435";
+        let h520 = b"0x012345678910111213141516171819202122232425262728293031323334353";
+
+        let tests = vec![
+            (PactType::any_try_into(h32),  Ok(PactType::StringLike(StringLike(h32)))),
+            (PactType::any_try_into(h64),  Ok(PactType::StringLike(StringLike(h64)))),
+            (PactType::any_try_into(h128), Ok(PactType::StringLike(StringLike(h128)))),
+            (PactType::any_try_into(h160), Ok(PactType::StringLike(StringLike(h160)))),
+            (PactType::any_try_into(h256), Ok(PactType::StringLike(StringLike(h256)))),
+            (PactType::any_try_into(h264), Ok(PactType::StringLike(StringLike(h264)))),
+            (PactType::any_try_into(h512), Ok(PactType::StringLike(StringLike(h512)))),
+            (PactType::any_try_into(h520), Ok(PactType::StringLike(StringLike(h520)))),
+        ];
+        for (lhs, rhs) in tests {
+            assert_eq!(lhs, rhs);
+        }
+    }
+
+    #[test]
+    fn it_converts_numeric_associated_types() {
+        trait Foo {
+            type Number32;
+            type Number64;
+        }
+
+        struct Bar;
+
+        impl Foo for Bar {
+            type Number32 = u32;
+            type Number64 = u64;
+        }
+
+        let n32: <Bar as Foo>::Number32 = 10u32;
```

Comment:
💭 Wondering does it work if we don't specify `10u32` i.e. just `10`?

Reply:
Yep it does work because `10` gets coerced into `u32` (that's `Number32`).
But `(PactType::any_try_into(&10)` will produce `PactConversionErr::UnknownType` error because it defaults to `i32`, which we don't have implemented 😄

Reply:
Perfect.